File: Searching/Linear-Search.py
def search(l,to_search):
  for e in l:
    if e==to_search:
      print("Position of %d is:"%to_search,l.index(e)+1)
      break
  else:print('Position Not Found For ',to_search)
lis,x=[1,5,6,3,-5,-4,0,2,-14,45],0
search(lis,x)
search(lis,100)

File: Searching/linear-search.cpp
//program to implement the linear search algorithm

#include<bits/stdc++.h>
using namespace std;

int search(int arr[], int n, int k) {
	for(int i=0;i<n;i++){
		if(arr[i]==k){
	            return i+1;
	        }
	}
	return -1;
}
int main()
{
    int arr[]={0,1,1,3,5,6,8,20,20,35,46,89};
    int k=35;
    int n= sizeof(arr)/sizeof(arr[0]);
    
    int index= search(arr,n,k);
    if(index==-1)
    {
        cout<<"Number not found\n";
    }
    else
    {
        cout<<"Number "<<k<<" is at index "<<index<<endl;
    }
    return 0;
}

